data/provider/src/main/java/tech/pegasys/teku/api/ChainDataProvider.java

```diff
@@ -689,6 +693,31 @@ protected int calculateProposerSyncAggregateBlockRewards(
     return total;
   }
 
+  @VisibleForTesting
+  protected UInt64 calculateProposerSlashingsRewards(
+      final BeaconBlock beaconBlock,
+      final tech.pegasys.teku.spec.datastructures.state.beaconstate.BeaconState state) {
+    final SszList<ProposerSlashing> proposerSlashings =
+        beaconBlock.getBody().getProposerSlashings();
+
+    UInt64 proposerSlashingsRewards = ZERO;
+    for (ProposerSlashing slashing : proposerSlashings) {
+      final UInt64 epoch = spec.computeEpochAtSlot(state.getSlot());
+      final SpecConfig specConfig = spec.getSpecConfig(epoch);
+
+      final int slashedIndex = slashing.getHeader1().getMessage().getProposerIndex().intValue();
+      final Validator validator = state.getValidators().get(slashedIndex);
+
+      final UInt64 whistleblowerReward =
+          validator.getEffectiveBalance().dividedBy(specConfig.getWhistleblowerRewardQuotient());
+      final UInt64 proposerReward =
+          whistleblowerReward.dividedBy(specConfig.getProposerRewardQuotient());
+      proposerSlashingsRewards = proposerSlashingsRewards.plus(proposerReward);
```

Comment:
https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/beacon-chain.md#proposerslashing
to me it looks like, because whistleblower index is proposer index, the proposer gets 2 increases to balance...
so they get 
 - proposer reward, which is `(whistleblower_reward / proposer_Reward_quotient)`
 - whistleblower_reward which is `(validator.effective_balance / whistleblower_reward_quotient)`
 
 maps to `BeaconStateMutators.private void slashValidator(MutableBeaconState state, int slashedIndex, int whistleblowerIndex)`